compiler/rustc_typeck/src/check/coercion.rs

```diff
@@ -1590,16 +1590,27 @@ impl<'tcx, 'exprs, E: AsCoercionSite> CoerceMany<'tcx, 'exprs, E> {
         let hir::ExprKind::Loop(_, _, _, loop_span) = expr.kind else { return;};
         let mut span: MultiSpan = vec![loop_span].into();
         span.push_span_label(loop_span, "this might have zero elements to iterate on");
+        let mut i = 0;
         for ret_expr in ret_exprs {
             span.push_span_label(
                 ret_expr.span,
                 "if the loop doesn't execute, this value would never get returned",
             );
+            i = i + 1;
+            if i == 3 {
```

Comment:
Instead of this you can use the `take` function on iterators and down below you could compare with the limit instead of `i`. To prevent duplicating the limit, you could use a constant.